src/cli.ts

```diff
@@ -117,10 +118,58 @@ function debugCliInfo(version: string, channel: string, env: Env, config: IConfi
   );
 }
 
+class VersionObject {
+  public cliVersion: string;
+  public pluginVersions: string[];
+  public osVersion: string;
+
+  public constructor(cliVersion: string, pluginVersions: string[], osVersion: string) {
+    this.cliVersion = cliVersion;
+    this.pluginVersions = pluginVersions;
+    this.osVersion = osVersion;
+  }
+}
+
+class SfdxMain extends Main {
+  // Function which returns Version object which includes cli version, plugin versions, OS and its version.
+  protected getIssueVersionObject(): VersionObject {
+    const cliVersion = this.config.userAgent;
+    const pluginVersion: string = exec('sfdx plugins --core', {
+      silent: true,
+    }).toString();
+    const pluginVersions: string[] = pluginVersion.split('\n');
+    pluginVersions.pop();
+    const osVersion = `${os.type()} ${os.release()}`;
+    return new VersionObject(cliVersion, pluginVersions, osVersion);
+  }
+
+  protected _version(): never {
```

Comment:
How does this get called?  Some kind of cool oclif magic?

Reply:
@amphro have to answer this.